### PR TITLE
KEYCLOAK-4980 SAML adapter should return 403 when unauthenticated Aja…

### PR DESCRIPTION
**Pull request by hokuda**
…x client accesses

### CODE REVIEW

**Comment by stianst** [comment]
This should be done in the same was as "autodetect-bearer-only" is done for OIDC adapter.

**Comment by hokuda** [comment]
@stianst thanks for your comment. I have a customer who wants SAML adapter to behave as PicketLink does. Do you have any other idea for it?

**Comment by stianst** [comment]
The autodetect-bearer-only option for OIDC adapter if added to SAML adapter would do exactly what this PR does though and more. Wouldn't it?

**Comment by hokuda** [comment]
@stianst @hmlnarik please review this updated PR. thanks.

**Review by hmlnarik — CHANGES_REQUESTED** [review]
Thank you, the PR looks almost good to me and brings SAML adapter on par with OIDC in this respect. Please see the comments inline.

**Comment by hmlnarik** [comment]
@hokuda Thank you for the related PRs, they both look good to me. Are you planning to write the tests or would you prefer me to write them?

**Comment by hokuda** [comment]
@hmlnarik Thank you for reviewing them! I am willing to write tests but have some problem resolving dependencies to build. Please let me try again today, but, if no success, I prefer you to write them.

**Comment by hokuda** [comment]
@hmlnarik Please reivew the changes made according to your suggestion. I wrote the test in testsuite/integration/src/test/java/org/keycloak/testsuite/keycloaksaml/SamlAdapterTest.java, since a same test for OIDC adapter is there in testsuite/integration/src/test/java/org/keycloak/testsuite/adapter/AdapterTest.java.

**Comment by hmlnarik** [comment]
@hokuda Thank you. Sadly, the tests need to go to AbstractSAMLServletsAdapterTest for adapters to be tested. Not only testsuite/integration is obsoleted by testsuite/integration-arquillian but they are also not tested with wildfly/eap because of the old testsuite way of testing (that's one of the reasons for deprecating the old testsuite).

**Comment by hokuda** [comment]
@hmlnarik OK, I copied the test to testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/adapter/servlet/AbstractSAMLServletsAdapterTest.java. Thanks.

**Comment by hmlnarik** [comment]
@hokuda Thank you very much for the PR. Can you please remove the changes to old testsuite (testsuite/integration) and squash the commits into just one? Then it will be ready for merging from my perspective.

**Comment by hokuda** [comment]
@hmlnarik rebase done!

**Comment by hmlnarik** [comment]
@hokuda Great job, thank you

**Comment by stianst** [comment]
Thanks @hokuda

**Comment by hokuda** [comment]
@stianst @hmlnarik Thanks a lot for your assistance and merging this!
